lib/repo_tools.pm

```diff
@@ -493,4 +494,51 @@ sub parse_repo_data {
     return \%repo_data;
 }
 
+=head2 verify_software
+
+ verify_software(%args);
+
+Validates that package or pattern is installed, or not installed and/or if
+package is available in the given repo.
+returns string with error or empty string in case of matching expectations.
+C<%args> should have following keys defined:
+- C<name>: package or pattern name
+- C<installed>: if set to true, validate that package or pattern is installed
+- C<pattern>: set to true if is pattern, otherwise validating package
+- C<available>: if set to true, validate that package or pattern is available in
+                the list of packages with given search criteria, otherwise
+                expect zypper command to fail
+- C<repo>: Optional, name of the repo where the package should be available. Check
+           is triggered only if C<available> is set to true
+
+=cut
+
+sub verify_software {
+    my (%args) = @_;
+
+    my $zypper_args = $args{installed} ? '--installed-only' : '--not-installed-only';
+    # define search type
+    $zypper_args .= $args{pattern} ? ' -t pattern' : ' -t package';
+    # Negate condition if package should not be available
+    my $cmd = $args{available} ? '' : '! ';
+    $cmd .= "zypper --non-interactive se -n $zypper_args --match-exact --details @{[ $args{name} ]}";
```

Comment:
`name` seems not to be an expression (is the key that we iterate), so we don't need this special operator to interpolate it.

Reply:
I've kept the structure which allows to use method for both scenarios. Using this method it's clear which package is not matching expectations. In case we will get enormous list of packages to validate, we can refactor this code to be faster and check multiple packages at a time.